### ISSUE TITLE
bug: Foundry deploy causes bug due to ast property being undefined

### ISSUE THREAD

**Issue by dragonsisdumb** [scaffold-eth/scaffold-eth-2#805]
### Is there an existing issue for this?

- [X] I have looked through the [existing issues](https://github.com/scaffold-eth/scaffold-eth-2/issues)

### Which method was used to setup Scaffold-ETH 2 ?

npx create-eth@latest

### Current Behavior

If you create a new scaffold-eth2 project using `foundry`, whilst deploying (without touching any of the code apart from `packages/foundry/.env` file to add `DEPLOYER_PRIVATE_KEY`), you will get an error at the end of the deploy, like this

```
TypeError: Cannot read properties of undefined (reading 'nodes')
    at getInheritedFromContracts (/Users/.../packages/foundry/script/generateTsAbis.js:40:38)
    at getInheritedFunctions (/Users/.../packages/foundry/script/generateTsAbis.js:53:34)
    at /Users/.../packages/foundry/script/generateTsAbis.js:109:29
    at Array.forEach (<anonymous>)
    at /Users/.../packages/foundry/script/generateTsAbis.js:101:24
    at Array.forEach (<anonymous>)
    at main (/Users/.../packages/foundry/script/generateTsAbis.js:93:10)
    at Object.<anonymous> (/Users/.../packages/foundry/script/generateTsAbis.js:143:3)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
```

### Expected Behavior

No error about `ast` property being undefined

### Steps To Reproduce

* `npx create-eth@latest`
* choose foundry
* run `yarn chain`
* run `yarn start`
* run `yarn deploy`

### Anything else?

Suggest the following edit to `packages/foundry/script/generateTsAbis.js`

```
function getInheritedFromContracts(artifact) {
  let inheritedFromContracts = [];
  if('ast' in artifact) {
    for (const astNode of artifact.ast.nodes) {
      if (astNode.nodeType == "ContractDefinition") {
        if (astNode.baseContracts.length > 0) {
          inheritedFromContracts = astNode.baseContracts.map(
            ({ baseName }) => baseName.name
          );
        }
      }
    }
  }
  return inheritedFromContracts;
}
```

**Comment by technophile-04** [comment]
Hey thanks @dragonsisdumb, for raising this 

> whilst deploying (without touching any of the code apart from packages/foundry/.env file to add DEPLOYER_PRIVATE_KEY)

Could please elaborate on this more? I just did : 

1. `npx create@latest` => selected foundry => yes for install dependency 
2. Manually updated `DEPLOYER_PRIVATE_KEY` with random private key in `packages/foundry/.env`  
3. `yarn chain` => `yarn deploy`

And it working fine for me. 

> Suggest the following edit to packages/foundry/script/generateTsAbis.js

Thanks for this ! Nevertheless, I think it is worth adding that `if(artifact.ast)` , do you want to tackle this @dragonsisdumb ? 

You could create an PR against [`cli`](https://github.com/scaffold-eth/scaffold-eth-2/tree/cli) branch, by updating this file https://github.com/scaffold-eth/scaffold-eth-2/blob/c55d25803fec3d11adf817cc87d8c04b3fcb0ad7/templates/extensions/foundry/packages/foundry/script/generateTsAbis.js#L39

Here is the [contributing guide](https://github.com/scaffold-eth/scaffold-eth-2/blob/cli/contributors/DEVELOPER-GUIDE.md#developer-guide), but I think changes are straightforward, please let us know if you need any help 🙌

**Comment by dragonsisdumb** [comment]
> You could create an PR against [cli](https://github.com/scaffold-eth/scaffold-eth-2/tree/cli) branch, by updating this file

Yep, threw up a PR

**Comment by technophile-04** [comment]
Fixed at #806
